From fe130a6a242d15c52fc7206125ff7863f0e3f174 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 18 Jan 2023 13:50:30 +0100
Subject: [PATCH 245/954] feat(stm32mp2-fdts): add HASH, PKA and SAES

Add nodes in device tree for HASH, PKA and SAES peripherals. HASH is
available for all STM32MP2 SoCs, so add it in stm32mp251.dtsi. PKA and
SAES are only available for STM32MP25xC or STM32MP25xF versions. Add the
nodes in the corresponding dtsi files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d60386a08c38220f8ae1abf3c478d1184fec1257 feat(stm32mp2-fdts): add HASH, PKA and SAES]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ia0f035aa5caad22ae76680a7d610815f952731ef
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/285739
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 fdts/stm32mp251.dtsi  |  8 ++++++++
 fdts/stm32mp25xc.dtsi | 17 +++++++++++++++++
 fdts/stm32mp25xf.dtsi | 17 +++++++++++++++++
 3 files changed, 42 insertions(+)

diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index aa909814f4..50a3ba5bb3 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -257,6 +257,14 @@
 			};
 		};
 
+		hash: hash@42010000 {
+			compatible = "st,stm32mp13-hash";
+			reg = <0x42010000 0x400>;
+			clocks = <&rcc CK_BUS_HASH>;
+			resets = <&rcc HASH_R>;
+			status = "disabled";
+		};
+
 		rng: rng@42020000 {
 			compatible = "st,stm32mp13-rng";
 			reg = <0x42020000 0x400>;
diff --git a/fdts/stm32mp25xc.dtsi b/fdts/stm32mp25xc.dtsi
index 8afb68583d..de85a94982 100644
--- a/fdts/stm32mp25xc.dtsi
+++ b/fdts/stm32mp25xc.dtsi
@@ -5,4 +5,21 @@
  */
 
 / {
+	soc@0 {
+		saes: saes@42050000 {
+			compatible = "st,stm32-saes";
+			reg = <0x42050000 0x400>;
+			clocks = <&rcc CK_BUS_SAES>;
+			resets = <&rcc SAES_R>;
+			status = "disabled";
+		};
+
+		pka: pka@42060000 {
+			compatible = "st,stm32-pka64";
+			reg = <0x42060000 0x2000>;
+			clocks = <&rcc CK_BUS_PKA>;
+			resets = <&rcc PKA_R>;
+			status = "disabled";
+		};
+	};
 };
diff --git a/fdts/stm32mp25xf.dtsi b/fdts/stm32mp25xf.dtsi
index 8afb68583d..de85a94982 100644
--- a/fdts/stm32mp25xf.dtsi
+++ b/fdts/stm32mp25xf.dtsi
@@ -5,4 +5,21 @@
  */
 
 / {
+	soc@0 {
+		saes: saes@42050000 {
+			compatible = "st,stm32-saes";
+			reg = <0x42050000 0x400>;
+			clocks = <&rcc CK_BUS_SAES>;
+			resets = <&rcc SAES_R>;
+			status = "disabled";
+		};
+
+		pka: pka@42060000 {
+			compatible = "st,stm32-pka64";
+			reg = <0x42060000 0x2000>;
+			clocks = <&rcc CK_BUS_PKA>;
+			resets = <&rcc PKA_R>;
+			status = "disabled";
+		};
+	};
 };
-- 
2.53.0

